content/blog/2017/2017-02-14-declarative-notifications.adoc

```diff
@@ -0,0 +1,437 @@
+I've used a new branch
+link:https://github.com/bitwiseman/hermann[my same fork] of the
+link:https://github.com/reiseburo/hermann[hermann project]:
+link:https://github.com/bitwiseman/hermann/tree/blog/declarative/notifications[`blog/add-declarative/notifications`].
+I've set up a Multibranch Pipeline and pointed it at my repository
+the same as did it previous post.
+Also the same as before, I've set this Pipeline's Git configuration to
+automatically "Clean after checkout".
+
+In addition, I still have my targets setup from the
+"link:/blog/2016/07/18/pipline-notifications/[Sending Notifications in Pipeline]" blog post.
+Take a look there for help setting up the
+plugin:slack[Slack],
+plugin:hipchat[HipChat],
+and plugin:email-ext[Email-ext]
+plugins.
+
+
+Here's the final form of the Declarative Pipeline from the previous post:
+
+[pipeline]
+----
+// Declarative //
+pipeline {
+  agent {
+    // Use docker container
+    docker {
+      image 'ruby:2.3'
+    }
+  }
+  options {
+    // Only keep the 10 most recent builds
+    buildDiscarder(logRotator(numToKeepStr:'10'))
+  }
+  stages {
+    stage ('Install') {
+      steps {
+        // install required gems
+        sh 'bundle install'
+      }
+    }
+    stage ('Build') {
+      steps {
+        // build
+        sh 'bundle exec rake build'
+      }
+
+      post {
+        success {
+          // Archive the built artifacts
+          archive includes: 'pkg/*.gem'
+        }
+      }
+    }
+    stage ('Test') {
+      steps {
+        // run tests with coverage
+        sh 'bundle exec rake spec'
+      }
+
+      post {
+        success {
+          // publish html
+          publishHTML target: [
+              allowMissing: false,
+              alwaysLinkToLastBuild: false,
+              keepAll: true,
+              reportDir: 'coverage',
+              reportFiles: 'index.html',
+              reportName: 'RCov Report'
+            ]
+        }
+      }
+    }
+  }
+  post {
+    always {
+      echo "Send notifications for result: ${currentBuild.result}"
+    }
+  }
+}
+// Scripted //
+----
+
+
+== Preparing for Notifications
+
+The Pipeline above has a final `always` block that prints a message to the status log,
+but doesn't print anything at the start of the run.
+The final form of the Pipeline in the original blog post called the same method twice,
+once at the start and once at the end of the run.
+Let's start by doing that here as well.
+
+[source, groovy]
+----
+pipeline {
+  /* ... unchanged ... */
+  stages {
+    stage ('Start') {
+      steps {
+        // send build started notifications
+        sendNotifications 'STARTED'
+      }
+    }
+    /* ... unchanged ... */
+  }
+  post {
+    always {
+      sendNotifications currentBuild.result
+    }
+  }
+}
+
+def sendNotifications(String buildStatus = 'STARTED') {
+  echo "Send notifications for run: ${buildStatus}"
+}
+----
+
+.Declarative is still Jenkins Pipeline 
```

Comment:
@rtyler - 
This is meant as an aside, 
In the process of writing theses blogs I feel like I've finally figured out a perspective on Declarative the makes sense to me and I wan to communicate that here.    Technically, I could just drop this entire section - but I wrote and then deleted a similar section for the previous post, which indicate to me that it is worth bringing up. 

I understand we're under time pressure, so if this seems off in the weeds, feel free to nuke it.  I needed to finish it, that's all.